Automattic/jetpack
# Update/recommendations summary navigation

#### Changes proposed in this Pull Request:

* This PR adds some extra navigation controls to the recommendations flow to allow individual recommendation steps to be viewed by clicking on the recommendation name on the summary screen.
* On individual recommendation steps, if the recommended feature has been enabled, the recommendation screen now shows "Enabled" and provides a link to related settings for the feature.
* If the summary screen has already been viewed, individual recommendation steps also provide a link back to the summary screen.

#### Jetpack product discussion


pbNhbs-1iw-p2#comment-7724

#### Does this pull request change what data or activity we track or use?


Yes, this PR removes one analytics event and adds several new ones.
On the recommendations summary screen, the following analytics event is removed (it used to get recorded when clicking on the name of one of the recommendations in the summary list):
```
jetpack_recommendations_summary_learn_more_click
```

The following analytics events are added:
```
/* Recommendations summary screen */
// Recorded when clicking on the name of a recommended feature or resource with the feature slug passed as a prop
jetpack_recommendations_summary_step_name_click


/*  Individual feature recommendation */
// If a feature is already enabled, this is recorded when the "settings" button for the feature is clicked, feature slug passed as a prop
jetpack_recommended_feature_configure_click

// Recorded when clicking "View Summary" from the recommendation step, feature slug passed as a prop
jetpack_recommended_feature_back_to_summary_click


/* Individual resource recommendation */
// Recorded when clicking "View Summary" from the recommendation step, feature slug passed as a prop
jetpack_recommended_resource_back_to_summary_click

```

#### Testing instructions:






Before starting, it's helpful to clear some option values. If you are using the docker development environment, you can run these commands using `jetpack docker`.
* `wp jetpack options delete recommendations_data`
* `wp jetpack options delete recommendations_step`
* Reload the site after clearing the options.

After clearing options:
* Navigate to `/wp-admin/admin.php?page=jetpack#/recommendations/site-type`
* Proceed through the recommendations flow, note that when viewing individual feature recommendations, each step provides a CTA to activate the feature or to skip it:

![Screen Shot 2022-04-19 at 4 56 15 PM](https://user-images.githubusercontent.com/18016357/164106350-9c03b737-36c8-478d-9e50-143e9e2281bb.png)

* When you reach the summary screen, you should now be able to click on the names of each feature/ resource to view that step in the flow again.

![Screen Shot 2022-04-19 at 5 40 42 PM](https://user-images.githubusercontent.com/18016357/164106515-84490abd-5d67-451d-8b21-d7f798ceef2f.png)

* After you have reached the summary screen and go back to view an individual step, you should have a new control below the CTA button to "View Summary". Clicking "View Summary" should return you to the summary screen.

![Screen Shot 2022-04-19 at 4 11 26 PM](https://user-images.githubusercontent.com/18016357/164106592-4dd7e2fe-7472-4296-8cc1-56ecc9aa2f8b.png)

* If you did not activate any of the features in the flow, activate "Downtime Monitor" from the summary screen.
* After Downtime Monitor is activated, click on the feature name in the summary screen to view the individual recommendation step.
* Notice that, since the feature is active, an "Enabled" label shows on the step and the CTA link now links to the settings for the feature. Additionally, the skip link below the CTA button now reads "Next" instead of "Not Now". 

![Screen Shot 2022-04-19 at 5 05 30 PM](https://user-images.githubusercontent.com/18016357/164106656-3ea2b930-d0e7-463c-8df9-d33d78b58653.png)

* Test activating a few other features and confirm that the CTA buttons on the individual recommendations are linking to the proper settings controls for each feature.
* Return to the summary screen. If you do not have a "Resources" section, follow the test plan on #23600 to activate a conditional recommendation for Site Security. (TLDR: you can enable and disable the hello dolly plugin to activate the recommendation).
* Test that you can also click on the names of items in the "Resources" list to view their respective recommendation steps.
